app/controllers/AdditionalMetadataClosureStatusController.scala

```diff
@@ -51,7 +52,7 @@ class AdditionalMetadataClosureStatusController @Inject() (
           if (consignment.files.nonEmpty) {
             val filePaths = consignment.files.flatMap(_.fileMetadata).filter(_.name == clientSideOriginalFilepath).map(_.value)
             val areAllFilesClosed = consignmentService.areAllFilesClosed(consignment)
-            cache.set(s"$consignmentId-data", (consignment.consignmentReference, filePaths, details.parentFolderId.get), 1.hour)
+            cache.set(s"$consignmentId-data", (consignment.consignmentReference, filePaths, closureProperties), 1.hour)
```

Comment:
I'd be tempted to move the caching into the display properties services, on the `getDisplayProperties` function.
Can then set the cache name based metadata type passed in, or simply cache the whole set of properties an filter the result based on the metadata type.

Reply:
We are already using the cache at some other places as well. Since we have to cache some other data too along with display properties, so it's better to use it at the controller level.